__tests__/app.test.js

```diff
@@ -7,143 +7,215 @@ const app = require("../app");
 afterAll(() => db.end());
 beforeEach(() => seed(testData));
 
-describe("GET /api/topics", () => {
-  test("200: returns an array", async () => {
-    const res = await request(app).get("/api/topics").expect(200);
-    expect(res.body.length).toBe(3);
-    res.body.forEach((topic) => {
-      expect(topic).toEqual({
-        slug: expect.any(String),
-        description: expect.any(String),
+describe("TOPICS", () => {
+  describe("GET /api/topics", () => {
+    test("200: returns an array", async () => {
+      const res = await request(app).get("/api/topics").expect(200);
+      expect(res.body.length).toBe(3);
+      res.body.forEach((topic) => {
+        expect(topic).toMatchObject({
+          slug: expect.any(String),
+          description: expect.any(String),
+        });
       });
     });
-  });
-  test("404: incorrect path returns 'path not found'", async () => {
-    await request(app).get("/api/topic").expect(404);
+    test("404: incorrect path returns 'path not found'", async () => {
+      await request(app).get("/api/topic").expect(404);
+    });
   });
 });
 
-describe.only("GET /api/article/:article_id", () => {
-  test("200: returns copy of requested article object including comment count", async () => {
-    const res = await request(app).get("/api/articles/9").expect(200);
-    expect(res.body.article).toEqual({
-      author: "butter_bridge",
-      title: "They're not exactly dogs, are they?",
-      article_id: 9,
-      body: "Well? Think about it.",
-      topic: "mitch",
-      created_at: expect.any(String),
-      votes: 0,
-      comment_count: 2,
+describe("ARTICLES", () => {
+  describe("GET /api/articles", () => {
+    test("200: returns array of article objects", async () => {
+      const res = await request(app).get("/api/articles").expect(200);
+      expect(res.body.articles).toHaveLength(12);
+      res.body.articles.forEach((article) => {
+        expect(article).toMatchObject({
+          author: expect.any(String),
+          title: expect.any(String),
+          article_id: expect.any(Number),
+          topic: expect.any(String),
+          created_at: expect.any(String),
+          votes: expect.any(Number),
+          comment_count: expect.any(Number),
+        });
+      });
     });
-  });
-  test("200: returns copy of requested article object which has zero comments", async () => {
-    const res = await request(app).get("/api/articles/2").expect(200);
-    expect(res.body.article).toEqual({
-      author: "icellusedkars",
-      title: "Sony Vaio; or, The Laptop",
-      article_id: 2,
-      body: expect.any(String),
-      topic: "mitch",
-      created_at: expect.any(String),
-      votes: 0,
-      comment_count: 0,
+    test("200: articles are sorted by date order descending", async () => {
+      const res = await request(app).get("/api/articles").expect(200);
+      expect(res.body.articles).toBeSorted({
+        key: "created_at",
+        descending: true,
+        coerce: true,
```

Comment:
Oooh, I've not seen the `coerce` property on this object before! Rather than ask our dear Pail Shrelington about it, would you please be my teacher for a moment and tell me what this does?

Reply:
So it checks the sorting numerically.. I think!

Reply:
That's cool! I'm going to give this a google myself!